core/src/avm2/bytearray.rs

```diff
@@ -185,53 +185,57 @@ impl ByteArrayStorage {
     }
 
     /// Compress the ByteArray into a temporary buffer
-    pub fn compress<'gc>(
-        &mut self,
-        algorithm: CompressionAlgorithm,
-    ) -> Result<Vec<u8>, Error<'gc>> {
+    pub fn compress(&mut self, algorithm: CompressionAlgorithm) -> Vec<u8> {
         let mut buffer = Vec::new();
-        match algorithm {
+        let error: Option<Box<dyn std::error::Error>> = match algorithm {
             CompressionAlgorithm::Zlib => {
-                let mut compresser = ZlibEncoder::new(&*self.bytes, Compression::fast());
-                compresser.read_to_end(&mut buffer)?;
+                let mut encoder = ZlibEncoder::new(&*self.bytes, Compression::fast());
+                encoder.read_to_end(&mut buffer).err().map(|e| e.into())
             }
             CompressionAlgorithm::Deflate => {
-                let mut compresser = DeflateEncoder::new(&*self.bytes, Compression::fast());
-                compresser.read_to_end(&mut buffer)?;
+                let mut encoder = DeflateEncoder::new(&*self.bytes, Compression::fast());
+                encoder.read_to_end(&mut buffer).err().map(|e| e.into())
             }
             #[cfg(feature = "lzma")]
-            CompressionAlgorithm::Lzma => lzma_rs::lzma_compress(&mut &*self.bytes, &mut buffer)?,
+            CompressionAlgorithm::Lzma => lzma_rs::lzma_compress(&mut &*self.bytes, &mut buffer)
+                .err()
+                .map(|e| e.into()),
             #[cfg(not(feature = "lzma"))]
-            CompressionAlgorithm::Lzma => {
-                return Err("Ruffle was not compiled with LZMA support".into())
-            }
+            CompressionAlgorithm::Lzma => Some("Ruffle was not compiled with LZMA support".into()),
+        };
+        if let Some(error) = error {
```

Comment:
Should we be throwing an error here?

Reply:
Right. I wrongly assumed that if `ByteArray.compress` just returns an empty array without throwing, so does `ByteArray.uncompress`. I fixed that.